provider/netcup.go

```diff
@@ -80,28 +80,46 @@ func (p *NetcupProvider) Records(ctx context.Context) ([]*endpoint.Endpoint, err
 			// some information is on DNS zone itself, query it first
 			zone, err := p.session.InfoDnsZone(domain)
 			if err != nil {
-				return nil, fmt.Errorf("unable to query DNS zone info for domain '%v': %v", domain, err)
+				p.logger.ErrorContext(ctx, "unable to query DNS zone info for domain", "domain", domain, "error", err.Error())
+				continue
+				//return nil, fmt.Errorf("unable to query DNS zone info for domain '%v': %v", domain, err)
 			}
 			ttl, err := strconv.ParseUint(zone.Ttl, 10, 64)
 			if err != nil {
-				return nil, fmt.Errorf("unexpected error: unable to convert '%s' to uint64", zone.Ttl)
+				p.logger.ErrorContext(ctx, "unable to parse TTL for domain", "domain", domain, "error", err.Error())
+				continue
+				//return nil, fmt.Errorf("unexpected error: unable to convert '%s' to uint64", zone.Ttl)
 			}
 			// query the records of the domain
 			recs, err := p.session.InfoDnsRecords(domain)
 			if err != nil {
 				if p.session.LastResponse != nil && p.session.LastResponse.Status == string(nc.StatusError) && p.session.LastResponse.StatusCode == 5029 {
-					p.logger.Debug("no records exist", "domain", domain, "error", err.Error())
+					p.logger.InfoContext(ctx, "no records exist", "domain", domain, "error", err.Error())
+					continue
 				} else {
-					return nil, fmt.Errorf("unable to get DNS records for domain '%v': %v", domain, err)
+					p.logger.ErrorContext(ctx, "unable to get DNS records for domain", "domain", domain, "error", err.Error())
+					continue
+					//return nil, fmt.Errorf("unable to get DNS records for domain '%v': %v", domain, err)
 				}
 			}
-			p.logger.Info("got DNS records for domain", "domain", domain)
+			p.logger.InfoContext(ctx, "got DNS records for domain", "domain", domain)
+
+			// TODO: move this into a separate function, so it can be covered in tests
```

Comment:
This could be done first, to cover it with tests before doing the changes. I may consider that when splitting this PR.

Reply:
That sounds good. Ideally split the PR early so I can review the easy bits quickly :)